Libraries/StyleSheet/ColorPropType.js

```diff
@@ -0,0 +1,34 @@
+ /**
+  * Copyright (c) 2015-present, Facebook, Inc.
+  * All rights reserved.
+  *
+  * This source code is licensed under the BSD-style license found in the
+  * LICENSE file in the root directory of this source tree. An additional grant
+  * of patent rights can be found in the PATENTS file in the same directory.
+  *
+  * @providesModule ColorPropType
+  */
+'use strict';
+var ReactPropTypes = require('ReactPropTypes');
+var tinycolor = require('tinycolor');
+
+var colorValidator = function (props, propName) {
+  var selectedColor = props[propName];
+  if (selectedColor === null || selectedColor === undefined || selectedColor.toString().trim() === '') {
+    return new Error(
+      `Invalid argument supplied to ${propName}.Expected a string like #123ADF or 'red'.`
+    );
+  }
+
+  if (tinycolor(selectedColor.toString().trim()).isValid()) {
```

Comment:
It seems like you are duplicating a lot of the same logic as in processColor ( https://github.com/facebook/react-native/blob/0e8b207cc34f230eb2cced3981734b522b601817/Libraries/StyleSheet/processColor.js#L17 ). Could you instead make processColor return either the correct color or `null`. Then here you would just check if it's not null.

This way they always stay in sync.